### ISSUE TITLE
fix: force role holder can participate out of turn in `LlamaRelativeUniqueHolderQuorum`

### ISSUE THREAD

**Issue by 0xrajath** [llamaxyz/llama#467]
https://github.com/spearbit-audits/review-llama2/pull/1#discussion_r1277715759

TL;DR:

You're saying that the force role doesn't follow the `creationTime-1` values which generally ensures that the role quantities that can be used are snapshotted at Action creation time. So anyone can be granted a force role in the middle of this action and then get to single handedly push the (dis)approval through even though they didn't have it at creationTime-1.

But this is only applicable to LlamaRelativeUniqueHolderQuorum strategy since that's the only one that has `if (forceApprovalRole[role]) return type(uint96).max;`.

All other strategies have, `return quantity > 0 && forceApprovalRole[role] ? type(uint96).max : quantity;` where quantity is from `creationTime-1` and `quantity>0` means they had that role at that time.
